### PR TITLE
HPCC-20719 Fix 2 problems in ESP Transaction Logging

### PR DESCRIPTION
**Pull request by wangkx**
1. the filterLogContent() should not overwrite original logging
request because other logging agent may use it;
2. wsloggingserviceagent.xsl should only create the setting for
one agent instance because multiple agent instances have been
handled by loggingmanager related xsl.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
I just created this PR to replace https://github.com/hpcc-systems/HPCC-Platform/pull/11854 which has been approved by @asselitx.
@richardkchapman please merge this PR with https://github.com/hpcc-systems/LN/pull/895

File: esp/logging/logginglib/loggingagentbase.cpp [hunk 1]
```diff
@@ -294,8 +294,8 @@ void CLogContentFilter::filterLogContent(IEspUpdateLogRequestWrap* req)
     StringBuffer updateLogRequestXML;
     toXML(updateLogRequestTree, updateLogRequestXML);
     ESPLOG(LogMax, "filtered content and option: <%s>", updateLogRequestXML.str());
-    req->clearOriginalContent();
-    req->setUpdateLogRequest(updateLogRequestXML.str());
+
+    return new CUpdateLogRequestWrap(req->getGUID(), req->getOption(), updateLogRequestXML.str());
```
Comment (richardkchapman) [inline esp/logging/logginglib/loggingagentbase.cpp]: This whole bit of code feels very heavyweight for logging...

Comment (wangkx) [reply esp/logging/logginglib/loggingagentbase.cpp]: I am not sure that I understand the comment. Are you saying that I should not create a new CUpdateLogRequestWrap?  The filterLogContent() may be called by multiple logging agents in different threads. If each of them does not have their copy of logging request, one logging agent may change the original request and another agent may not see the original request.

Comment (wangkx) [reply esp/logging/logginglib/loggingagentbase.cpp]: The logging function is designed to handle multiple logging agents per application. It should also handle different kinds of logging requests (xml string, ptree, and one mixed group for now). A user configures some filter to filter the requests before the requests are queued. If one logging attempt is failed, the request has to be re-queued (not in this method). A user may also configure some mapping to pick the parts of the requests to send out (not in this method). When I have time, some parts of the code do need to be optimized.

**Comment by richardkchapman** [comment]
@afishbeck Can you look at this and see if you agree regarding logging weight

**Pull request merged**